splash/resources.py

```diff
@@ -201,9 +201,27 @@ def _log_stats(self, request, options):
             "timestamp": int(time.time()),
             "user-agent": (request.getHeader(b"user-agent").decode('utf-8')
                            if request.getHeader(b"user-agent") else None),
-            "args": repr(options)
         }
-        log.msg(json.dumps(msg), system="events")
+        msg.update(self.detailed_options(options))
+        try:
+            msg = json.dumps(msg)
+        except TypeError:
+            # splash-as-proxy options contain byte objects which are not
+            # serializable
+            pass
+        log.msg(msg, system="events")
+
+    def detailed_options(self, options):
+        url = options.get("url")
+        if not url:
+            return options
+        url = six.moves.urllib.parse.urlparse(url)
+        options.update({
+            "netloc": url.netloc,
+            "query": url.query,
+            "path": url.path
+        })
```

Comment:
I agree with @kmike , `url` could be anything. I would rather do the extra parsing outside of Splash, in log analyzers, like Logstash for example, which can be extended to do URL parsing, or may have that already.

I believe leaving this PR's purpose of making parsing of arguments easier as proper JSON is already a big win, without analyzing them fully at this stage.